schema.ts

```diff
@@ -1064,6 +1064,68 @@ export type ObjectSchemaPathProperties = Pick<
   | 'popularityRankProperty'
 >;
 
+/**
+ * Specifies how this property should be indexed.
+ * @hidden
+ */
+export enum IndexingStrategy {
+  Standard = 'standard',
+  Raw = 'raw',
+}
+
+/**
+ * A list of properties that will be used to provide context when indexing a property for full-text search.
+ * @hidden
+ */
+export type ContextProperties = Array<PropertyIdentifier<string>>;
+
+export type BasicIndexedProperty = PropertyIdentifier<string>;
+
+/**
+ * A property to be indexed, supporting a more detailed definition than {@link BasicIndexedProperty}.
+ * TODO(alexd): Unhide this
+ * @hidden
+ */
+export interface DetailedIndexedProperty {
+  /**
+   * A property to be indexed. Must be a {@link ValueType.String} property.
+   */
+  property: PropertyIdentifier<string>;
+  /**
+   * The strategy to be used for indexing this property.
+   */
+  strategy: IndexingStrategy;
+}
+
+export type IndexedProperty = BasicIndexedProperty | DetailedIndexedProperty;
+
+/**
+  * Defines how to index objects for use with full-text indexing.
+  * TODO(alexd): Unhide this
+  * @hidden 
+  */
+export interface IndexDefinition {
+  /**
+   * A list of properties from within {@link ObjectSchemaDefinition.properties} that should be indexed.
+   */
+  properties: IndexedProperty[];
+  
+  /*
+   * The context properties to be used for indexing.
+   * If unspecified, defaults to the default context properties for the schema.
```

Comment:
What are the default context properties for the schema?

Reply:
Changed this comment.